### ISSUE TITLE
Folders?

### ISSUE THREAD

**Issue by eemantsal** [wallabag/wallabag#3350]
When one starts to have dozens or hundreds of saved pages would be handy to be abe to organize them. Could you make a filesystem structure so we can classify our saved items in an ordered way?
If you don't consider folders to be a good idea, what about a tag view, that in the end would make a similar function? With "tag view" I mean to have a panel at the left with all our tags ordered alphabetically or how the user wants (size, age...) and when we clic on ony all the items tagged with that concrete tag are presented. Just like a file browser but if you don't want folders, with tags.
I don't have a real preference, but my items list is a damn mess and needs some order, and I suppose many other people's is too :-|

**Comment by nicosomb** [comment]
Hello @eemantsal 
Tags are already implemented in wallabag. 
It's not planned to implement folder structure. 

Please reopen this issue if necessary.

**Comment by eemantsal** [comment]
Yes, I know that tags are already implemented, I'm a heavy user of them. But didn't meant that.
Let me explain better since I'm aware I explained myself horribly in my opening message.

What I meant was an option to present the saved pages' list or thumbnails as in a species of file browser based on tags instead on a directory structure (since you have said don't like folders) and make it available as the default view. Now the default view is https://yourwallabagserver.xyz/unread/list, I propose to let the user to choose that the default view be https://yourwallabagserver.xyz/tag/list.

But I'd also add some other stuff.
By now, if you click on the "Tags" menu at the left panel, you get at the right a two columns view of all your tags, and when you click on one, those tags columns disappear and the corresponding thumbnails is shown.
Well, I'd put that tags menu in a narrow one column dropdown submenu with all the tags, so you cpuld easily show/hide that tags column and scroll down/up to localize the tagged of pages you want.
Look at this screenshot from Evernote's interface (old, I believe, but that's the idea):
![serveimage](https://user-images.githubusercontent.com/18057272/30516103-7f1dabf8-9b35-11e7-8687-787dc2cdb5d7.png)
(In the screenshot the button with the arrow opens the list of notebooks in Evernote, but the same could be implemented for tags in Wallabag.)

Then, you would click the tag, say, "photography" and the list of "photography" tagged pages would be shown in a second lateral panel at the right (like that piece of right panel you can see in the previous screenshot); and then, you would clic the desired thumbnail/list item to open and read the page in a thrid and bigger panel at the very right.
Again, pretty much like in Evernote (imagine that instead a "Notes" item is selected in the left panel, there's a tags list), but Kmail and others permit to place tags for mail as well as a folder structure:
https://zapier.cachefly.net/storage/photos/f9d9e81c1e62caa236eb667dcbfa7ef3.png

Another option could be: 
Left panel with the "Tags" menu as it is now | 2º panel (narrow) with the tags list in a single colum | 3º Panel (wide) with the list/thumbnails of taggd pages
When one would click an item in that 3º panel the thumbnails would let their place to the saved page view. Well, I suppose you can think of some other behaviors maybe even better

Besides, I think this type of layuod would make management easier, because one could select several pages, and drag & drop them onto any tag in the tags column so the article gets automatically tagged accordingly. This would probably require a pop up dialog that asked the user if wants the new tag to be added to that/those that the pages already have or if wants to completely substitute.

I hope I've been more explanatory now. I don't know if I'm suggesting a too hard to do feature, but I hopw you like the idea.

**Comment by douglasg14b** [comment]
Tags are a poor solution for organization by itself... Tags augment structure, they don't replace it. 

Tags are messy, disorganized, and are contextual information to enable you to find related topics. Lists/Categories/Folders are a primary organizational structure that are augmented by tags, they exist to reduce cognitive overhead when trying to find something. Tags, by themselves, reintroduce that same cognitive overhead that lists/structure attempts to resolve. 

When you have hundreds of tags, how do you at a glance, find a collection of articles relating to a subtopic that you wanted to group together? You can't, instead you need to sift through a swarm of tags to try and find that niche one you made to try and organize that collection of articles.

**Comment by techexo** [comment]
Indeed, but if you need to have an organized collection of documents with folders and tags, maybe wallabag's not the right tool?

In my opinion, wallabag is really to read articles later (or offline after sync in an app), not to store things on the long term. In that case, I would use something like Zotero or any organized content manager.
